herbert_core/classes/MPIFramework/@JobDispatcher/private/send_tasks_to_workers_.m

```diff
@@ -36,93 +36,99 @@
+
+            job_info = mf.initial_framework_info;
+            cluster_wrp.finalize_all(); % will destroy current mf
+            % Reinitialize mf and create job folder
+            mf = MessagesFilebased(job_info);
+            if ~isempty(pc.shared_folder_on_local)
+                mf.mess_exchange_folder = pc.shared_folder_on_local;
+            end
+
+            obj.mess_framework_ = mf;
+
+            cluster_wrp = par_fc.get_initialized_cluster(n_workers,mf);
+            [cluster_wrp,ok] = cluster_wrp.wait_started_and_report(obj.task_check_time);
+            ic= ic+1;
+        end
+        if ~ok
+            error('PARALLEL_FRAMEWORK:runtime_error',...
+                  ' Can not start parallel cluster %s after %d attempts. Parallel job aborted',...
+                  class(cluster_wrp),n_restart_attempts+1);
         end
-        
-        obj.mess_framework_ = mf;
-        %
-        cluster_wrp = par_fc.get_initialized_cluster(n_workers,mf);
-        [cluster_wrp,ok] = cluster_wrp.wait_started_and_report(obj.task_check_time);
-        ic= ic+1;
     end
-    if ~ok
-        error('PARALLEL_FRAMEWORK:runtime_error',...
-            ' Can not start parallel cluster %s after %d attempts. Parallel job aborted',...
-            class(cluster_wrp),n_restart_attempts+1);
+
+    if keep_workers_running % store cluster pointer for job resubmission
+        obj.cluster_       = cluster_wrp;
+        obj.job_destroyer_ = onCleanup(@()finalize_all(cluster_wrp));
+    else % clear cluster on exit
+        clob_mf = onCleanup(@()finalize_all(cluster_wrp));
     end
-end
 
-if keep_workers_running % store cluster pointer for job resubmission
-    obj.cluster_       = cluster_wrp;
-    obj.job_destroyer_ = onCleanup(@()finalize_all(cluster_wrp));
-else % clear cluster on exit
-    clob_mf = onCleanup(@()finalize_all(cluster_wrp));
-end
+    [outputs,n_failed,task_ids,obj] = submit_and_run_job_(obj,task_class_name,...
+                                                      common_params,loop_params,return_results,...
+                                                      cluster_wrp,keep_workers_running);
+    if exist('clob_mf', 'var')
+        clear clob_mf;
+    end
 
-[outputs,n_failed,task_ids,obj] = submit_and_run_job_(obj,task_class_name,...
-    common_params,loop_params,return_results,...
-    cluster_wrp,keep_workers_running);
-if exist('clob_mf', 'var')
-    clear clob_mf;
 end
 
-
 function n_wk = check_loop_param(loop_param,n_workers)
-n_wk = n_workers;
-if iscell(loop_param)
-    n_jobs = numel(loop_param);
-elseif isnumeric(loop_param)
-    n_jobs = loop_param;
-elseif isstruct(loop_param)
-    fn = fieldnames(loop_param);
-    par1 = loop_param.(fn{1});
-    if iscell(par1)
-        n_jobs = numel(par1);
+    % Available number of workers
+    n_wk = n_workers;
+
+    if ~isscalar(loop_param)
+        n_jobs = numel(loop_param);
+    elseif isscalar(loop_param) && isnumeric(loop_param)
```

Comment:
is the `isscalar` test required here? The initial `if` will have caught any `loop_param` that are not scalar